vllm/lora/request.py

```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class LoRARequest:
+    """
+    Request for a LoRA adapter.
+
+    Note that this class should be be used internally. For online
+    serving, it is recommended to not allow users to use this class but
+    instead provide another layer of abstraction to prevent users from
+    accessing unauthorized LoRA adapters.
+
+    lora_id and lora_int_id must be globally unique for a given adapter.
+    This is currently not enforced in vLLM.
+    """
+
+    lora_id: str
```

Comment:
It looks like this particular field is not used anywhere else in the engine code. However, there are a lot of references to `lora_id` as an integer, which is actually the `lora_int_id` field in this class.

The idea of using `hash(lora_id)` as `lora_int_id` is problematic due to the positivity requirement (i.e `hash()` can be negative/zero). I guess `abs(hash(s)) + 1` might be better, but that will collide sooner or later and cause trouble.

I think the code should either work with `lora_id` being int (essentially dropping existing `lora_id` and renaming `lora_int_id` to `lora_id`) OR work with a string `lora_id` + some additional bookkeeping that picks and remembers integer ids for the string identifier.

Reply:
The main reason for `lora_id` was debugging (you can give loras friendly names), but I agree it is somewhat confusing to have it. Perhaps the easiest path forward would be to actually rename the field to `lora_friendly_name`/`lora_name` and make it optional. That should make it clear that it's not being used for id purposes.

Reply:
Let's rename it to `lora_name`?